docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/front-end/src/lib/core/analytics/landing-page-contents-helper.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ todo: all notes from chats, keeping here until finalized
+
+	- passhomepage other
+	- function that gets homepage el
+	- find row, find parent element - that makes up that row
+		- promo block is promo block
+		- feature card is the row
+		- guide card could be anything and that's dev's problem not mine
+
+	- data points should all be the same, or close enough so
+
+	- row and some sort of helper functions and/or a number
+	- promoblock will just be row item 1
+
+	- dev controls field template
+	- wut r "row"
+		- section tag with usa-section?
+		- a div of a direct child?
+		- data-el on the element that defines row?
+			- might make life easier
+
+	- row of video as well, some element
+	- raw html
+	- add data element to clickable
+	- data-eddl in the html
+
+ 	Landing page contents analytics helper
+ */
+
+/**
+ * Gets "rows" for a landing page.
+ *
+ * Rows are currently defined as parents of components such as promo blocks,
+ * feature cards, guide cards, raw html blocks, etc. The data-eddl attribute
+ * is required on all elements that should be counted as a row.
+ *
+ * @example
+ * const rowsTotal = getPageRows().length;
+ *
+ * @returns {HTMLElement[]} Array of HTML element "rows" found in the document.
+ */
+export const getPageRows = (): HTMLElement[] => {
+	const rows = document.querySelectorAll('[data-eddl]');
```

Comment:
This should be more specific, as we have other `data-eddl-*` data attributes that are not home/landing.